modules/shortcodes/instagram.php

```diff
@@ -17,6 +17,65 @@
 use Automattic\Jetpack\Connection\Client;
 use Automattic\Jetpack\Constants;
 
+if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+	add_action( 'init', 'jetpack_instagram_enable_embeds' );
```

Comment:
> I'm still learning about things, is there a reason this can't be done for non-wpcom as well?

It might work on JP as well -- I haven't tried TBH :sweat_smile: 

I started out with the non-wrapped version (`add_filter()` statements at module level), which worked fine on JP, but not on WP.com -- which is why I introduced this check. I also drew inspiration from the Getty shortcode/embed: https://github.com/Automattic/jetpack/blob/a62c3e4472ce64cfd613e7945b73ab59f3ff769f/modules/shortcodes/getty.php#L11-L34

> It seems to me that it would be cleaner to not branch the code based on whether it's on wpcom or not, if that doesn't somehow break anything.

That's a fair point! However, it would also mean that JP and WP.com would be out of sync once again :sweat_smile: 
Would it be okay if we merge this PR as-is (to get them back into sync), and I'll file a new one to drop the `IS_WPCOM` guard (which we can then sync to WP.com via JP Fusion)? That would unblock syncing, and would allow us to experiment with the `IS_WPCOM` check in isolation (in case e.g. it turns out that it doesn't work on JP).

Reply:
> However, it would also mean that JP and WP.com would be out of sync once again :sweat_smile: 

Also a good point! Yeah, let's do what you suggest.